test/integration/acceptance/fake_peer_example_test.cpp

```diff
@@ -318,80 +330,98 @@ TEST_F(FakePeerExampleFixture,
 }
 
 /**
- * Check that after receiving a valid command the ITF peer provides a proposal
- * containing it.
+ * Check that ITF peer is able to get a proposal from a Fake peer and commit the
+ * transaction stored in it.
  *
  * \attention this code is nothing more but an example of Fake Peer usage
  *
- * @given a network of two iroha peers
- * @when a valid command is sent to one
- * @then it must either (on demand) provide a proposal containing this command,
- * or request it from the other peer
+ * @given a network of one real and one fake peers
+ * @when fake peer provides a proposal with valid tx
+ * @then the real peer must commit the transaction from that proposal
  */
 TEST_F(FakePeerExampleFixture,
        OnDemandOrderingProposalAfterValidCommandReceived) {
 
-  /* A custom behaviour that requests a proposal for the round it got vote for,
-   * and if gets one, checks that the proposal contains the given tx hash.
-   */
-  struct CustomBehaviour : public fake_peer::HonestBehaviour {
-    CustomBehaviour(const interface::types::HashType &tx_hash,
-                    std::atomic_flag &got_proposal_from_main_peer)
-        : tx_hash_(tx_hash),
-          got_proposal_from_main_peer_(got_proposal_from_main_peer) {}
-
-    void processYacMessage(fake_peer::YacMessagePtr message) override {
-      const auto proposal_from_main_peer = getFakePeer().sendProposalRequest(
-          message->front().hash.vote_round, kProposalWaitingTime);
-      if (proposal_from_main_peer
-          and std::any_of(proposal_from_main_peer->transactions().begin(),
-                          proposal_from_main_peer->transactions().end(),
-                          [this](const auto &tx) {
-                            return tx.reducedHash() == tx_hash_;
-                          })) {
-        got_proposal_from_main_peer_.test_and_set(std::memory_order_relaxed);
-      }
-      HonestBehaviour::processYacMessage(message);
-    }
-
-    const interface::types::HashType &tx_hash_;
-    std::atomic_flag &got_proposal_from_main_peer_;
-  };
-
   // Create the tx:
   const auto tx = complete(
       baseTx(kAdminId).transferAsset(kAdminId, kUserId, kAssetId, "tx1", "1.0"),
       kAdminKeypair);
-  const auto hash = tx.reducedHash();
 
-  std::atomic_flag got_proposal_from_main_peer = ATOMIC_FLAG_INIT;
+  auto state_ready = std::make_shared<std::atomic_bool>(false);
```

Comment:
I guess it's possible just passing a reference, w/o shared_ptr,

Reply:
Then it would have lifetime issues when the variable gets wiped from stack but still used by rx streams in fake peer on heap.